### PR TITLE
Feature/video external url

### PR DESCRIPTION
**Pull request by nuritsha**

### CODE REVIEW

File: includes/widgets/video.php [hunk 1]
```diff
@@ -182,6 +182,17 @@ protected function _register_controls() {
 			]
 		);
 
+		$this->add_control(
+			'inset_from_url',
```
Comment (KingYes) [inline includes/widgets/video.php]: insert

File: includes/widgets/video.php [hunk 2]
```diff
@@ -196,6 +207,23 @@ protected function _register_controls() {
...
+
+		$this->add_control(
+			'external_url',
+			[
+				'label' => __( 'URL', 'elementor' ),
+				'type' => Controls_Manager::TEXT,
+				'dynamic' => [
+					'active' => true,
+				],
+				'media_type' => 'video',
```
Comment (KingYes) [inline includes/widgets/video.php]: remove line

File: includes/widgets/video.php [hunk 3]
```diff
@@ -1025,13 +1055,15 @@ private function get_hosted_video_url() {
 	}
 
 	/**
+	 * @param bool $from_media
+	 *
 	 * @since 2.1.0
 	 * @access private
 	 */
-	private function render_hosted_video() {
+	private function render_hosted_video( $from_media ) {
```
Comment (KingYes) [inline includes/widgets/video.php]: remove param

**Review by KingYes — CHANGES_REQUESTED** [review]

File: includes/widgets/video.php [hunk 4]
```diff
@@ -997,13 +1018,16 @@ private function get_hosted_params() {
...
 	/**
+	 * @param bool $from_media
+	 *
+	 * @return string
 	 * @since 2.1.0
 	 * @access private
 	 */
-	private function get_hosted_video_url() {
+	private function get_hosted_video_url( $from_media ) {
 		$settings = $this->get_settings_for_display();
```
Comment (kobizz) [inline includes/widgets/video.php]: It's better to just send the requested url as parameter.

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by KingYes — APPROVED** [review]

**Pull request merged**